unix_integration/src/unix_proto.rs

```diff
@@ -16,6 +17,59 @@ pub struct NssGroup {
     pub members: Vec<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub enum PamMessageStyle {
+    #[default]
+    PamPromptEchoOff,
+    PamPromptEchoOn,
+    PamErrorMsg,
+    PamTextInfo,
+}
+
+impl PamMessageStyle {
+    pub fn value(&self) -> i32 {
+        match *self {
+            PamMessageStyle::PamPromptEchoOff => 1,
+            PamMessageStyle::PamPromptEchoOn => 2,
+            PamMessageStyle::PamErrorMsg => 3,
+            PamMessageStyle::PamTextInfo => 4,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub enum CredType {
+    #[default]
+    Password,
+    MFACode,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct PamPrompt {
+    pub style: PamMessageStyle,
+    pub msg: String,
+    pub timeout: Option<u64>, // timeout of None means use the config default
+    pub cred_type: Option<CredType>,
+}
+
+impl PamPrompt {
+    // Produce a typical password prompt
+    pub fn passwd_prompt() -> Self {
+        PamPrompt {
+            style: PamMessageStyle::PamPromptEchoOff,
+            msg: "Password: ".to_string(),
+            timeout: None,
+            cred_type: Some(CredType::Password),
+        }
+    }
```

Comment:
We could perhaps provide a default MFA prompt here, but nothing would use it at the moment.